### PR TITLE
JDK-8261969: SNIHostName should check if the encoded hostname conform to RFC 3490

### PR DESCRIPTION
**Pull request by johnshajiang**
Similar to the constructor SNIHostName(String hostname), the constructor SNIHostName(byte[] encoded) also needs to check if the encoded hostname conform to RFC 3490.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8261969](https://bugs.openjdk.java.net/browse/JDK-8261969): SNIHostName should check if the encoded hostname conform to RFC 3490

### Reviewers
 * [Rajan Halade](https://openjdk.java.net/census#rhalade) (@rhalade - **Reviewer**)
 * [Xue-Lei Andrew Fan](https://openjdk.java.net/census#xuelei) (@XueleiFan - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2639/head:pull/2639`
`$ git checkout pull/2639`

### CODE REVIEW

File: test/jdk/javax/net/ssl/ServerName/IllegalSNIName.java [hunk 1]
```diff
@@ -23,32 +23,50 @@
...
 import javax.net.ssl.SNIHostName;
+import java.nio.charset.StandardCharsets;
+import java.util.HexFormat;
 
 public class IllegalSNIName {
 
+    private static void checkHostname(String hostname) throws Exception {
+        try {
+            new SNIHostName(hostname);
+            throw new Exception("Expected to get IllegalArgumentException for "
```
Comment (rhalade) [inline test/jdk/javax/net/ssl/ServerName/IllegalSNIName.java:40]:
```suggestion
            throw new RuntimeException("Expected to get IllegalArgumentException for "
```

File: test/jdk/javax/net/ssl/ServerName/IllegalSNIName.java [hunk 2]
```diff
@@ -23,32 +23,50 @@
...
+                    + hostname);
+        } catch (IllegalArgumentException iae) {
+            // That's the right behavior.
+        }
+    }
+
+    private static void checkHostname(byte[] encodedHostname) throws Exception {
+        try {
+            new SNIHostName(encodedHostname);
+            throw new Exception("Expected to get IllegalArgumentException for "
```
Comment (rhalade) [inline test/jdk/javax/net/ssl/ServerName/IllegalSNIName.java:50]:
```suggestion
            throw new RuntimeException("Expected to get IllegalArgumentException for "
```

**Review by rhalade — CHANGES_REQUESTED** [review]

**Comment by johnshajiang** [comment]
I wouldn't merge the master to this branch :-(

**Comment by johnshajiang** [comment]
Could the latest patch be reviewed?

**Review by XueleiFan — APPROVED** [review]

**Review by rhalade — APPROVED** [review]
Thanks for updating the test.

**Comment by johnshajiang** [comment]
/integrate

@rhalade @XueleiFan Thanks for your reviews!

**Pull request closed without merging**